src/main/groovy/gl51/project/store/MemoryProductStorage.groovy

```diff
@@ -0,0 +1,44 @@
+package gl51.project.store
+
+class MemoryProductStorage implements  ProductStorage {
+
+	List<Product> productList = [] 
+	int id = 1
+	
+    @Override
+    int save(Product p) {
+		p.id = id
+        productList.add(p)
+		id += 1
+    	return p.id
+    }
+
+    @Override
+    void update(int id, Product p) {
+        Product product = this.getByID(id)
+        int indexOfProduct = productList.indexOf(product)
+
+         productList.add(indexOfProduct,p)
```

Comment:
Adding at a given index doesn't remove the Product at this index, it only shifts it.
You should remove the old one
```suggestion
         productList.add(indexOfProduct,p)
         productList.remove(indexOfProduct+1)
```

Reply:
Je suis curieux de voir comment ca peut passer dans les tests :+1: 
va falloir etre créatif, parce que tel que, ca ne marche pas